### PR TITLE
Move mpi-operator v2 to the top of the repository

### PR DESCRIPTION
**Pull request by tenzen-y**
Signed-off-by: Yuki Iwai <yuki.iwai.tz@gmail.com>

I moved the mpi-operator v2 to the top of the repository.

Fixes: #486 

/assign @alculquicondor @terrytangyuan

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -26,24 +26,15 @@ git clone https://github.com/${GITHUB_USER}/mpi-operator.git
...
-The newest iteration of the controller is in the module `github.com/kubeflow/mpi-operator/v2`.
-
 ## Run tests
 
 ### Unit and integration tests
 
-You can execute all the unit and integration tests via `make test`.
-
-If you only which to run the tests for the v2 controller, you can run `make test_v2`.
+You can execute all the unit and integration tests via `make test_v2`.
```
Comment (alculquicondor) [inline CONTRIBUTING.md:33]: rename to `make test`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -84,8 +75,7 @@ make images dev_manifest
 kubectl apply -k manifests/overlays/dev
 ```
 
-The image comes bundled with all the controller versions. For example, you can
-find the v1 controller binary at `/opt/mpi-operator.v1`.
+The image comes bundled v2beta1.
```
Comment (alculquicondor) [inline CONTRIBUTING.md:78]:
```suggestion
The controller handles MPIJobs with API version v2beta1 or newer.
```

Comment (tenzen-y) [reply CONTRIBUTING.md:78]: Thanks!

**Comment by tenzen-y** [comment]
@alculquicondor I addressed your review comments and squashed all commits to one.

**Comment by alculquicondor** [comment]
/lgtm

The e2e tests should give us high confidence on this change.

**Review by terrytangyuan — APPROVED** [review]

**Pull request merged**